src/rust/engine/fs/src/store.rs

```diff
@@ -405,10 +406,21 @@ impl Store {
       .to_boxed()
   }
 
-  fn materialize_file(&self, destination: PathBuf, digest: Digest) -> BoxFuture<(), String> {
+  fn materialize_file(
+    &self,
+    destination: PathBuf,
+    digest: Digest,
+    is_executable: bool,
+  ) -> BoxFuture<(), String> {
     self
       .load_file_bytes_with(digest, move |bytes| {
         File::create(&destination)
+          .and_then(|f| {
```

Comment:
Might be able to do one fewer syscall with https://stackoverflow.com/a/28673836 ? But probably too micro.